src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs

```diff
@@ -183,15 +183,12 @@ public static bool IsPathRooted(string path)
             return path.Length > 0 && path[0] == DirectorySeparatorChar;
         }
 
-        private static unsafe byte[] CreateCryptoRandomByteArray(int byteLength)
+        private static byte[] CreateCryptoRandomByteArray(int byteLength)
         {
             var arr = new byte[byteLength];
-            fixed (byte* buf = arr)
+            if (!Interop.Crypto.GetRandomBytes(arr, arr.Length))
```

Comment:
Another point of interest is that RAND_pseudo_bytes is deprecated according to http://openssl.org/docs/manmaster/crypto/RAND_bytes.html
> RAND_pseudo_bytes() has been deprecated. Users should use RAND_bytes() instead.